eclipse/che
# Add namespace name cleanup before applying

### What does this PR do?

Sometimes users may have names incompartible with K8S objects naming (simplest example is using email as a username),
so in that case user-defined namespace creation with names generated using templates like `abc-<username>` will fail. 

This PR adds checks for symbols and length validity, sanitizes username if required, and stores it into user preferences for further usage.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/17841

### How to test this PR?

Configure Che to use `<username>` placeholder namespace template. Register user with name containing symbol(s) not allowed in kubernetes names. Log in with that user and start workspace. Namespace with sanitized name should be created.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
